File: sdk/r/R/affiliates_advanced.R
#' AffiliatesAdvanced: advanced affiliate management with client_id
#' @param http HttpClient object
#' @return Resource object
#' @export
AffiliatesAdvancedResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "AffiliatesAdvancedResource"
  env
}

#' GET /api/affiliates_adv/{client_id}/info - Get affiliate account info
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getInfo <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/info", list(...))
}

#' GET /api/affiliates_adv/{client_id}/stats - Get affiliate stats
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getStats <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/stats", list(...))
}

#' GET /api/affiliates_adv/{client_id}/commission-plans - Get commission plans
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getCommissionPlans <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/commission-plans", list(...))
}

#' GET /api/affiliates_adv/{client_id}/vouchers - Get vouchers
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getVouchers <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/vouchers", list(...))
}

#' GET /api/affiliates_adv/{client_id}/commissions - Get commissions
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getCommissions <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/commissions", list(...))
}

#' GET /api/affiliates_adv/{client_id}/referrals - Get referred clients
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getReferrals <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/referrals", list(...))
}

#' GET /api/affiliates_adv/{client_id}/payouts - Get payout history
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getPayouts <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/payouts", list(...))
}

#' GET /api/affiliates_adv/{client_id}/campaigns - Get campaigns
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getCampaigns <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/campaigns", list(...))
}

#' GET /api/affiliates_adv/{client_id}/audit - Get activity log
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getAuditLog <- function(self, ...) {
  self$http$get("/api/affiliates_adv/{client_id}/audit", list(...))
}

#' POST /api/affiliates_adv/{client_id}/activate - Activate affiliate account
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
activate <- function(self, ...) {
  self$http$post("/api/affiliates_adv/{client_id}/activate", list(...))
}

#' POST /api/affiliates_adv/{client_id}/commission-plan/{commission_id} - Set commission plan
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
setCommissionPlan <- function(self, ...) {
  self$http$post("/api/affiliates_adv/{client_id}/commission-plan/{commission_id}", list(...))
}

#' POST /api/affiliates_adv/{client_id}/vouchers/{plan_id} - Create voucher
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createVoucher <- function(self, ...) {
  self$http$post("/api/affiliates_adv/{client_id}/vouchers/{plan_id}", list(...))
}

#' DELETE /api/affiliates_adv/{client_id}/vouchers/{voucher_id} - Delete voucher
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteVoucher <- function(self, ...) {
  self$http$delete("/api/affiliates_adv/{client_id}/vouchers/{voucher_id}", list(...))
}

#' POST /api/affiliates_adv/{client_id}/landing-page - Set landing page
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
setLandingPage <- function(self, ...) {
  self$http$post("/api/affiliates_adv/{client_id}/landing-page", list(...))
}

File: sdk/c/src/affiliates_advanced.c
#include "hitechcloud/affiliates_advanced.h"
#include <stdlib.h>


/* GET /api/affiliates_adv/{client_id}/info - Get affiliate account info */
char *hc_affiliates_advanced_get_info(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/info", NULL);
}

/* GET /api/affiliates_adv/{client_id}/stats - Get affiliate stats */
char *hc_affiliates_advanced_get_stats(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/stats", NULL);
}

/* GET /api/affiliates_adv/{client_id}/commission-plans - Get commission plans */
char *hc_affiliates_advanced_get_commission_plans(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/commission-plans", NULL);
}

/* GET /api/affiliates_adv/{client_id}/vouchers - Get vouchers */
char *hc_affiliates_advanced_get_vouchers(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/vouchers", NULL);
}

/* GET /api/affiliates_adv/{client_id}/commissions - Get commissions */
char *hc_affiliates_advanced_get_commissions(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/commissions", NULL);
}

/* GET /api/affiliates_adv/{client_id}/referrals - Get referred clients */
char *hc_affiliates_advanced_get_referrals(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/referrals", NULL);
}

/* GET /api/affiliates_adv/{client_id}/payouts - Get payout history */
char *hc_affiliates_advanced_get_payouts(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/payouts", NULL);
}

/* GET /api/affiliates_adv/{client_id}/campaigns - Get campaigns */
char *hc_affiliates_advanced_get_campaigns(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/campaigns", NULL);
}

/* GET /api/affiliates_adv/{client_id}/audit - Get activity log */
char *hc_affiliates_advanced_get_audit_log(hc_http_client_t *client) {
    return hc_get(client, "/api/affiliates_adv/{client_id}/audit", NULL);
}

/* POST /api/affiliates_adv/{client_id}/activate - Activate affiliate account */
char *hc_affiliates_advanced_activate(hc_http_client_t *client) {
    return hc_post(client, "/api/affiliates_adv/{client_id}/activate", NULL);
}

/* POST /api/affiliates_adv/{client_id}/commission-plan/{commission_id} - Set commission plan */
char *hc_affiliates_advanced_set_commission_plan(hc_http_client_t *client) {
    return hc_post(client, "/api/affiliates_adv/{client_id}/commission-plan/{commission_id}", NULL);
}

/* POST /api/affiliates_adv/{client_id}/vouchers/{plan_id} - Create voucher */
char *hc_affiliates_advanced_create_voucher(hc_http_client_t *client) {
    return hc_post(client, "/api/affiliates_adv/{client_id}/vouchers/{plan_id}", NULL);
}

/* DELETE /api/affiliates_adv/{client_id}/vouchers/{voucher_id} - Delete voucher */
char *hc_affiliates_advanced_delete_voucher(hc_http_client_t *client) {
    return hc_delete(client, "/api/affiliates_adv/{client_id}/vouchers/{voucher_id}", NULL);
}

/* POST /api/affiliates_adv/{client_id}/landing-page - Set landing page */
char *hc_affiliates_advanced_set_landing_page(hc_http_client_t *client) {
    return hc_post(client, "/api/affiliates_adv/{client_id}/landing-page", NULL);
}

